### PR TITLE
Add BranchPythonVirtualenvOperator

### PR DESCRIPTION
**Pull request by jscheffl**
This PR is a follow-up and split of the PR https://github.com/apache/airflow/pull/33017 to add BranchPythonVirtualenvOperator.

Note: Initial planned decorators will be added in a follow-up PR.

FYI @AutomationDev85 @clellmann @wolfdn

### CODE REVIEW

File: airflow/operators/python.py [hunk 1]
```diff
@@ -623,6 +623,25 @@ def _iter_serializable_context_keys(self):
...
+    """
+    A workflow can "branch" or follow a path after the execution of this task in a virtualenv.
+    It derives the PythonVirtualenvOperator and expects a Python function that returns
+    a single task_id or list of task_ids to follow. The task_id(s) returned
+    should point to a task directly downstream from {self}. All other "branches"
+    or directly downstream tasks are marked with a state of ``skipped`` so that
+    these paths can't move forward. The ``skipped`` states are propagated
+    downstream to allow for the DAG state to fill up and the DAG run's state
+    to be inferred.
+    """
```
Comment (uranusjr) [inline airflow/operators/python.py]: Please follow the D205 rule, see #10742.

File: airflow/operators/python.py [hunk 2]
```diff
@@ -623,6 +623,25 @@ def _iter_serializable_context_keys(self):
...
+
+    def execute(self, context: Context) -> Any:
+        branch = super().execute(context)
+        self.log.info("Branch callable return %s", branch)
+        self.skip_all_except(context["ti"], branch)
+        return branch
```
Comment (uranusjr) [inline airflow/operators/python.py]: Can be a good idea to extract this into a `BranchMixin` so this does not get duplicated (three times with this addition).

File: airflow/operators/python.py (hunk 1 above)
Comment (jscheffl) [reply airflow/operators/python.py]: Applied review.

File: airflow/operators/python.py (hunk 2 above)
Comment (jscheffl) [reply airflow/operators/python.py]: Extracted as proposed.

**Comment by jens-scheffler-bosch** [comment]
Thanks @uranusjr for the early-bird review, now I cleaned up and added pytests... sorry took a while but now ready to review.

**Review by uranusjr — APPROVED** [review]
Nice.

**Review by potiuk — APPROVED** [review]

**Pull request merged**